src/mpi-base.jl

```diff
@@ -529,12 +529,48 @@ function Reduce{T}(object::T, op::Op, root::Integer, comm::Comm)
     isroot ? recvbuf[1] : nothing
 end
 
-function Scatter{T}(sendbuf::MPIBuffertype{T},
-                    count::Integer, root::Integer, comm::Comm)
+function Allreduce{T}(sendbuf::MPIBuffertype{T}, recvbuf::MPIBuffertype{T},
+                      count::Integer, op::Op, comm::Comm)
+    flag = Ref{Cint}()
+    ccall(MPI_ALLREDUCE, Void, (Ptr{T}, Ptr{T}, Ptr{Cint}, Ptr{Cint}, Ptr{Cint},
+          Ptr{Cint}, Ptr{Cint}), sendbuf, recvbuf, &count, &mpitype(T), 
+          &op.val, &comm.val, flag)
+
+    if flag[] != 0
```

Comment:
We don't check for errors in any other routine. By default, MPI aborts if an error is encountered, so that the check doesn't seem necessary. I'd rather omit it here as well for consistency.